src/lambertFetch.cpp

```diff
@@ -0,0 +1,310 @@
+    print( departureOrbitFile, departureOrbit, ephemerisFileHeader );
+
+    // Sample departure path.
+    const StateHistory departurePath = sampleKeplerOrbit( departureState,
+                                                          timeOfFlight,
+                                                          input.outputSteps,
+                                                          earthGravitationalParameter,
+                                                          departureEpoch );
+    // Write sampled departure path to file.
+    std::ostringstream departurePathFilePath;
+    departurePathFilePath << input.outputDirectory << "/transfer" << input.transferId << "_"
+                          << input.departurePathFilename;
+    std::ofstream departurePathFile( departurePathFilePath.str( ).c_str( ) );
+    print( departurePathFile, departurePath, ephemerisFileHeader );
+    departurePathFile.close( );
+
+    Vector6 arrivalState;
+    arrivalState[ 0 ] = arrivalPositionX;
+    arrivalState[ 1 ] = arrivalPositionY;
+    arrivalState[ 2 ] = arrivalPositionZ;
+    arrivalState[ 3 ] = arrivalVelocityX;
+    arrivalState[ 4 ] = arrivalVelocityY;
+    arrivalState[ 5 ] = arrivalVelocityZ;
+
+    // Compute period of arrival orbit.
+    const Vector6 arrivalStateKepler
+        = astro::convertCartesianToKeplerianElements( arrivalState,
+                                                      earthGravitationalParameter );
+    const double arrivalOrbitalPeriod
+        = astro::computeKeplerOrbitalPeriod( arrivalStateKepler[ astro::semiMajorAxisIndex ],
+                                             earthGravitationalParameter );
+
+    // Sample arrival orbit.
+    const StateHistory arrivalOrbit = sampleKeplerOrbit( arrivalState,
+                                                         arrivalOrbitalPeriod,
+                                                         input.outputSteps,
+                                                         earthGravitationalParameter,
+                                                         departureEpoch );
+
+    // Write sampled arrival orbit to file.
+    std::ostringstream arrivalOrbitFilePath;
+    arrivalOrbitFilePath << input.outputDirectory << "/transfer" << input.transferId << "_"
+                         << input.arrivalOrbitFilename;
+    std::ofstream arrivalOrbitFile( arrivalOrbitFilePath.str( ).c_str( ) );
+    print( arrivalOrbitFile, arrivalOrbit, ephemerisFileHeader );
+    arrivalOrbitFile.close( );
+
+    // Sample arrival path.
+    const StateHistory arrivalPath = sampleKeplerOrbit( arrivalState,
+                                                        -timeOfFlight,
+                                                        input.outputSteps,
+                                                        earthGravitationalParameter,
+                                                        departureEpoch + timeOfFlight );
```

Comment:
Well spotted! Since you corrected the error, can you post the correction in a comment here? Then I can add it to the PR.